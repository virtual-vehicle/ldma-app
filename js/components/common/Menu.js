import React, { Component } from 'react';
import {
  Dimensions,
  StyleSheet,
  ScrollView,
  View,
  Text,
  TouchableOpacity,
} from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Email } from 'react-native-openanything';
import { get as safeGet } from 'lodash';
import DeviceInfo from 'react-native-device-info';
import { COLORS } from '../../constants/colors';
import { logout } from '../../actions/userActions';
import {
  GO_TO_MAIN,
  GO_TO_SAFE_DRIVING,
  GO_TO_MY_TRIPS,
  GO_TO_RANKINGS,
} from '../../actions/actionTypes';

const WINDOW_WIDTH = Dimensions.get('window').width;
const WINDOW_HEIGHT = Dimensions.get('window').height;

type Props = {
  logout: any,
  navigation: any,
};

export class Menu extends Component<Props> {
  onSupportClick = (emailTo) => {
    const { user } = this.props;
    const username = safeGet(user, 'username', '');
    const email = emailTo;
    const subject = 'LDMA: Feedback/Questions';
    const device = `${DeviceInfo.getBrand()} ${DeviceInfo.getModel()}`;
    const osVersion = `${DeviceInfo.getSystemName()} ${DeviceInfo.getSystemVersion()}`;
    const appVersion = DeviceInfo.getReadableVersion();
    const body = `Please explain your feedback or questions as detailed as you can, thank you!

    ----
    Device: ${device}
    OS version: ${osVersion}
    User: ${username}
    App version: ${appVersion}`;

    Email(email, subject, body).catch(err => console.error(err));
  }

  userLogout(e) {
    e.preventDefault();
    const { logout } = this.props;
    logout();
  }

  render() {
    const { navigation, user } = this.props;
    const username = safeGet(user, 'username', '');

    const appVersion = DeviceInfo.getReadableVersion();
    return (
      <ScrollView scrollsToTop={false} style={styles.container}>
        <View style={styles.versionNameSection}>
          <Text style={styles.versionText}>VERSION {appVersion}</Text>
          <Text style={styles.nameText}>{username}</Text>
        </View>
        <View style={styles.content}>
          <Text
            style={styles.item}
            onPress={() => navigation.dispatch({ type: GO_TO_MAIN })}
          >
            TripViz ->
          </Text>
          <View style={styles.lineSeparator} />
            <Text
              style={styles.item}
              onPress={() => navigation.dispatch({ type: GO_TO_SAFE_DRIVING })}
            >
              Safe Driving ->
            </Text>
          <View style={styles.lineSeparator} />
            <Text
              style={styles.item}
              onPress={() => navigation.dispatch({ type: GO_TO_MY_TRIPS })}
            >
              My Trips ->
            </Text>
          <View style={styles.lineSeparator} />
            <Text
              style={styles.item}
              onPress={() => navigation.dispatch({ type: GO_TO_RANKINGS })}
            >
              My Rankings ->
            </Text>
          <View style={styles.lineSeparator} />
          <Text
            onPress={(e) => this.userLogout(e)}
            style={styles.item}
          >
            Log Out
          </Text>
          <View style={styles.lineSeparator} />
          <View>
            <Text style={styles.headlineText}>Customer support</Text>
            <TouchableOpacity
              onPress={() => this.onSupportClick('alexander.stocker@v2c2.at')}
            >
              <Text style={styles.emailText}>alexander.stocker@v2c2.at</Text>
            </TouchableOpacity>
            <Text style={[styles.headlineText, { marginTop: 20 }]}>Tech support</Text>
            <TouchableOpacity
              onPress={() => this.onSupportClick('nik.adzic@v2c2.at')}
            >
              <Text style={styles.emailText}>nik.adzic@v2c2.at</Text>
            </TouchableOpacity>
          </View>
        </View>

      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: WINDOW_WIDTH + 1,
    height: WINDOW_HEIGHT,
    backgroundColor: COLORS.LIGHTGREY4,
    borderRightWidth: 1,
    borderRightColor: COLORS.RED,
    borderStyle: 'solid',
  },
  versionNameSection: {
    backgroundColor: COLORS.LIGHTGREY,
    flex: 1,
    justifyContent: 'flex-end',
    width: '80%',
    height: 90,
    paddingLeft: 18,
    paddingBottom: 14,
  },
  nameText: {
    color: COLORS.WHITE,
    fontSize: 18,
  },
  versionText: {
    color: COLORS.WHITE,
    fontSize: 13,
  },
  item: {
    fontSize: 18,
  },
  content: {
    paddingLeft: 18,
    paddingRight: (WINDOW_WIDTH * 0.33) + 18,
    paddingTop: 36,
  },
  lineSeparator: {
    borderBottomColor: COLORS.GREY,
    borderBottomWidth: 0.5,
    marginTop: 19.5,
    marginBottom: 20,
  },
  headlineText: {
    color: COLORS.GREY2,
    fontSize: 14,
  },
  emailText: {
    color: COLORS.LIGHTGREY,
    fontSize: 16,
    fontWeight: 'bold',
  },
});

const mapStateToProps = (state) =>
  ({
    user: state.auth.user,
  });

const mapDispatchToProps = (dispatch) =>
  bindActionCreators({
    logout,
  }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Menu);
