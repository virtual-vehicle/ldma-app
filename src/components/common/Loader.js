import React, { Component } from 'react';
import {
  View,
  StyleSheet,
} from 'react-native';
import { Bubbles } from 'react-native-loader';
import { COLORS } from '../../constants/colors';

type Props = {
  white: boolean,
};

class Loader extends Component<Props> {
  render() {
    const { white } = this.props;
    return (
      <View style={styles.loading}>
        <Bubbles
          size={10}
          color={white ? COLORS.WHITE : COLORS.BLACK}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  loading: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'transparent',
  },
});

export default Loader;

