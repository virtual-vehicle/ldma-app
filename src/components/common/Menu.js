import React, { Component } from 'react';
import {
  Dimensions,
  StyleSheet,
  ScrollView,
  View,
  Text,
  Image,
  TouchableOpacity,
} from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import email from 'react-native-email';
import { get as safeGet } from 'lodash';
import DeviceInfo from 'react-native-device-info';
import { COLORS } from 'ldmaapp/src/constants/colors';
import { logout } from 'ldmaapp/src/actions/userActions';
import { selectNavigationItem } from 'ldmaapp/src/actions/navigationActions';
import NavigationService from 'ldmaapp/src/utils/navigation';

const WINDOW_WIDTH = Dimensions.get('window').width;
const WINDOW_HEIGHT = Dimensions.get('window').height;

type Props = {
  logout: any,
  selectNavigationItem: any,
  navigation: any,
  selected_navigation_item: String,
};

export class Menu extends Component<Props> {
  onSupportClick = (emailTo) => {
    const { user } = this.props;
    const username = safeGet(user, 'username', '');
    const to = emailTo;
    const subject = 'LDMA: Feedback/Questions';
    const device = `${DeviceInfo.getBrand()} ${DeviceInfo.getModel()}`;
    const osVersion = `${DeviceInfo.getSystemName()} ${DeviceInfo.getSystemVersion()}`;
    const appVersion = DeviceInfo.getReadableVersion();
    const body = `Please explain your feedback or questions as detailed as you can, thank you!

    ----
    Device: ${device}
    OS version: ${osVersion}
    User: ${username}
    App version: ${appVersion}`;

    // it does not work in emulator/simulator, because apple does not allow it.
    email(to, {
      // Optional additional arguments
      subject: subject,
      body: body,
    }).catch(console.error)
  }

  userLogout(e) {
    e.preventDefault();
    const { logout, selectNavigationItem } = this.props;
    selectNavigationItem('Home');
    logout();
  }

  navigate(navItem) {
    const { selectNavigationItem } = this.props;
    selectNavigationItem(navItem);
    NavigationService.navigate(navItem);
  }

  render() {
    const { user, selected_navigation_item } = this.props;
    const username = safeGet(user, 'username', '');

    const appVersion = DeviceInfo.getReadableVersion();
    return (
      <ScrollView scrollsToTop={false} style={styles.container}>
        <View style={styles.userSection}>
          <Image style={styles.userImage} source={require('ldmaapp/assets/png/userpic.png')} />
          <View>
            <Text style={styles.nameText}>{username}</Text>
          </View>
        </View>
        <View style={styles.content}>
          <View style={styles.lineSeparator} />
          <View style={styles.itemContainer}>
            <Image style={styles.itemImage} source={require('ldmaapp/assets/png/home.png')} />
            <Text
              style={selected_navigation_item === 'Home' ? styles.itemTextSelected : styles.itemText}
              onPress={() => this.navigate('Home')}
            >
              Home
            </Text>
          </View>
          <View style={styles.itemContainer}>
            <Image style={styles.itemImage} source={require('ldmaapp/assets/png/trips.png')} />
            <Text
              style={selected_navigation_item === 'MyTrips' ? styles.itemTextSelected : styles.itemText}
              onPress={() => this.navigate('MyTrips')}
            >
              My Trips
            </Text>
          </View>
          <View style={styles.itemContainer}>
            <Image style={styles.itemImage} source={require('ldmaapp/assets/png/ranking.png')} />
            <Text
              style={selected_navigation_item === 'Rankings' ? styles.itemTextSelected : styles.itemText}
              onPress={() => this.navigate('Rankings')}
            >
              Rankings
            </Text>
          </View>
          <View style={styles.lineSeparator} />
          <View style={styles.itemContainer}>
            <Image style={styles.itemImage} source={require('ldmaapp/assets/png/logout.png')} />
            <Text
              onPress={(e) => this.userLogout(e)}
              style={styles.itemText}
            >
              Logout
          </Text>
          </View>
          <View style={styles.lineSeparator} />
          <View>
            <Text style={styles.headlineText}>Customer support</Text>
            <TouchableOpacity
              onPress={() => this.onSupportClick('alexander.stocker@v2c2.at')}
            >
              <Text style={styles.emailText}>alexander.stocker@v2c2.at</Text>
            </TouchableOpacity>
            <Text style={[styles.headlineText, { marginTop: 20 }]}>Tech support</Text>
            <TouchableOpacity
              onPress={() => this.onSupportClick('nik.adzic@v2c2.at')}
            >
              <Text style={styles.emailText}>nik.adzic@v2c2.at</Text>
            </TouchableOpacity>
            <Text style={[styles.headlineText, { marginTop: 20 }]}>Version {appVersion}</Text>
          </View>
        </View>

      </ScrollView >
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: WINDOW_WIDTH + 1,
    height: WINDOW_HEIGHT,
    backgroundColor: COLORS.LIGHTGREY4,
    borderRightWidth: 1,
    borderRightColor: COLORS.RED,
    borderStyle: 'solid',
  },
  userSection: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    width: '80%',
    height: 100,
    paddingTop: 25,
    paddingLeft: 25,
    paddingBottom: 25,
    paddingRight: (WINDOW_WIDTH * 0.33) + 25,
  },
  userImage: {
    height: 50,
    width: 50,
    resizeMode: "contain",
    borderRadius: 50,
    marginRight: 25,
  },
  nameText: {
    color: COLORS.GREY2,
    fontSize: 18,
  },
  content: {
    paddingLeft: 25,
    paddingRight: (WINDOW_WIDTH * 0.33) + 25,
  },
  itemContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 31.5,
  },
  itemImage: {
    height: 25,
    width: 25,
    resizeMode: "contain",
  },
  itemText: {
    fontSize: 18,
    marginLeft: 25,
  },
  itemTextSelected: {
    color: COLORS.RED,
    fontSize: 18,
    marginLeft: 25,
  },
  lineSeparator: {
    borderBottomColor: COLORS.GREY,
    borderBottomWidth: 0.5,
    marginBottom: 31.5,
  },
  headlineText: {
    color: COLORS.GREY2,
    fontSize: 14,
  },
  emailText: {
    color: COLORS.LIGHTGREY,
    fontSize: 16,
    fontWeight: 'bold',
  },
});

const mapStateToProps = (state) =>
  ({
    user: state.auth.user,
    selected_navigation_item: state.navigation.selected_navigation_item,
  });

const mapDispatchToProps = (dispatch) =>
  bindActionCreators({
    logout,
    selectNavigationItem,
  }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Menu);
