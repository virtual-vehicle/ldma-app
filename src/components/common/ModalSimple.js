
import React from 'react';
import {
  View,
  TouchableOpacity,
  Text,
  StyleSheet,
} from 'react-native';
import Modal from 'react-native-modal';
import { COLORS } from 'ldmaapp/src/constants/colors';

const FIELDS_WIDTH = '88%';
const BUTTON_HEIGHT = 50;

const ModalSimple = (props) => {
  const {
    isVisible,
    onPress,
    headline,
    content,
  } = props;

  return (
    <Modal
      isVisible={isVisible}
      onBackdropPress={() => onPress()}
      style={styles.modalContainer}
    >
      <View style={styles.modalMainSection}>
        {headline && <Text style={styles.modalHeadlineText}>{headline}</Text>}
        {content && <Text style={styles.modalContentText}>{content}</Text>}
        <TouchableOpacity
          style={styles.modalButton}
          onPress={() => onPress()}
        >
          <Text style={styles.modalButtonText}>OK</Text>
        </TouchableOpacity>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  modalContainer: {
    display: 'flex',
    alignItems: 'center',
  },
  modalMainSection: {
    display: 'flex',
    alignItems: 'center',
    width: '100%',
    backgroundColor: COLORS.WHITE,
    borderRadius: 3,
  },
  modalHeadlineText: {
    color: COLORS.BLACK,
    fontSize: 24,
    textAlign: 'center',
    marginTop: 25,
    fontWeight: '500',
  },
  modalContentText: {
    marginTop: 17,
    fontSize: 18,
    textAlign: 'center',
    color: COLORS.GREY2,
  },
  modalButton: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: COLORS.BLUE,
    width: FIELDS_WIDTH,
    borderRadius: 5,
    height: BUTTON_HEIGHT,
    marginTop: 25,
    marginBottom: 30,
  },
  modalButtonText: {
    color: COLORS.WHITE,
    fontSize: 20,
  },
});

export default ModalSimple;

