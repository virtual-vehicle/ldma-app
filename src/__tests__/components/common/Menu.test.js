import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Menu } from 'ldmaapp/src/components/common/Menu';

Enzyme.configure({ adapter: new Adapter() });

const user = {
  email: 'test@email.com',
  auth_token: 'token',
};

describe('Menu component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Menu user={user} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('check the texts', () => {
    const wrapper = shallow(<Menu user={user} />);
    expect(wrapper.find('Text')).toHaveLength(11);
  });
});
