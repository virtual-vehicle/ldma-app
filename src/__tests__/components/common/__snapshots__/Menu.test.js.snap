// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component renders correctly 1`] = `
<ScrollViewMock
  scrollsToTop={false}
  style={
    Object {
      "backgroundColor": "#F1F1F1",
      "borderRightColor": "#e11027",
      "borderRightWidth": 1,
      "borderStyle": "solid",
      "flex": 1,
      "height": 1334,
      "width": 751,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#d1d1d1",
        "flex": 1,
        "height": 90,
        "justifyContent": "flex-end",
        "paddingBottom": 14,
        "paddingLeft": 18,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFF",
          "fontSize": 13,
        }
      }
    >
      VERSION 
    </Text>
    <Text
      style={
        Object {
          "color": "#FFF",
          "fontSize": 18,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "paddingLeft": 18,
        "paddingRight": 265.5,
        "paddingTop": 36,
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      TripViz -&gt;
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#cacaca",
          "borderBottomWidth": 0.5,
          "marginBottom": 20,
          "marginTop": 19.5,
        }
      }
    />
    <Text
      onPress={[Function]}
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Safe Driving -&gt;
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#cacaca",
          "borderBottomWidth": 0.5,
          "marginBottom": 20,
          "marginTop": 19.5,
        }
      }
    />
    <Text
      onPress={[Function]}
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      My Trips -&gt;
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#cacaca",
          "borderBottomWidth": 0.5,
          "marginBottom": 20,
          "marginTop": 19.5,
        }
      }
    />
    <Text
      onPress={[Function]}
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      My Rankings -&gt;
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#cacaca",
          "borderBottomWidth": 0.5,
          "marginBottom": 20,
          "marginTop": 19.5,
        }
      }
    />
    <Text
      onPress={[Function]}
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Log Out
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#cacaca",
          "borderBottomWidth": 0.5,
          "marginBottom": 20,
          "marginTop": 19.5,
        }
      }
    />
    <View>
      <Text
        style={
          Object {
            "color": "#4A4A4A",
            "fontSize": 14,
          }
        }
      >
        Customer support
      </Text>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#d1d1d1",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          alexander.stocker@v2c2.at
        </Text>
      </TouchableOpacity>
      <Text
        style={
          Array [
            Object {
              "color": "#4A4A4A",
              "fontSize": 14,
            },
            Object {
              "marginTop": 20,
            },
          ]
        }
      >
        Tech support
      </Text>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#d1d1d1",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          nik.adzic@v2c2.at
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</ScrollViewMock>
`;
